src/core/Akka.Remote/EndpointManager.cs

```diff
@@ -340,44 +373,62 @@ protected override SupervisorStrategy SupervisorStrategy()
-                                 ia.RemoteAddress, settings.RetryGateClosedFor.TotalMilliseconds, ia.Message);
-                        endpoints.MarkAsFailed(Sender, Deadline.Now + settings.RetryGateClosedFor);
-                        AddressTerminatedTopic.Get(Context.System).Publish(new AddressTerminated(ia.RemoteAddress));
+                        KeepQuarantinedOr(ia.RemoteAddress, () =>
+                        {
+                            var causedBy = ia.InnerException == null
+                                ? ""
+                                : string.Format("Caused by: [{0}]", ia.InnerException);
+                            _log.Warning("Tried to associate with unreachable remote address [{0}]. Address is now gated for {1} ms, all messages to this address will be delivered to dead letters. Reason: [{2}] {3}",
+                                ia.RemoteAddress, _settings.RetryGateClosedFor.TotalMilliseconds, ia.Message, causedBy);
+                            _endpoints.MarkAsFailed(Sender, Deadline.Now + _settings.RetryGateClosedFor);
+                        });
+
+                        if (ia.DisassociationInfo.HasValue && ia.DisassociationInfo == DisassociateInfo.Quarantined)
+                        {
+                            //TODO: add context.system.eventStream.publish(ThisActorSystemQuarantinedEvent(localAddress, remoteAddress))
+                        }
                         directive = Directive.Stop;
                     })
-                    .With<ShutDownAssociationException>(shutdown =>
+                    .With<ShutDownAssociation>(shutdown =>
                     {
-                        log.Debug("Remote system with address [{0}] has shut down. Address is now gated for {1}ms, all messages to this address will be delivered to dead letters.",
-                                  shutdown.RemoteAddress, settings.RetryGateClosedFor.TotalMilliseconds);
-                        endpoints.MarkAsFailed(Sender, Deadline.Now + settings.RetryGateClosedFor);
-                        AddressTerminatedTopic.Get(Context.System).Publish(new AddressTerminated(shutdown.RemoteAddress));
+                        KeepQuarantinedOr(shutdown.RemoteAddress, () =>
+                        {
+                            _log.Debug("Remote system with address [{0}] has shut down. Address is now gated for {1}ms, all messages to this address will be delivered to dead letters.",
+                                 shutdown.RemoteAddress, _settings.RetryGateClosedFor.TotalMilliseconds);
+                            _endpoints.MarkAsFailed(Sender, Deadline.Now + _settings.RetryGateClosedFor);
+                        });
                         directive = Directive.Stop;
                     })
-                    .With<HopelessAssociationException>(hopeless =>
+                    .With<HopelessAssociation>(hopeless =>
```

Comment:
`HopelessAssociation` - means a catastrophic failure occurred when attempting to connect to the remote node. No association was ever formed.